src/main/java/frc/robot/systems/GrabberFSM.java

```diff
@@ -0,0 +1,233 @@
+package frc.robot.systems;
+
+import com.revrobotics.CANSparkMax;
+
+import edu.wpi.first.wpilibj.Joystick;
+// Third party Hardware Imports
+// Robot Imports
+import frc.robot.TeleopInput;
+import frc.robot.HardwareMap;
+
+public class GrabberFSM {
+	/* ======================== Constants ======================== */
+	// FSM state definitions
+	public enum FSMState {
+		LOW_IDLE,
+		ASCENDING,
+        CONTROL,
+        DESCENDING,
+        MID_IDLE
+	}
+
+
+	/* ======================== Private variables ======================== */
+	private FSMState currentState;
+    private Joystick joystick;
```

Comment:
you should be using the shared TeleopInput class. take a look at how mech blue sets this up